cpmpy/solvers/cpo.py

```diff
@@ -0,0 +1,470 @@
+import shutil
+import warnings
+
+from .solver_interface import SolverInterface, SolverStatus, ExitStatus
+from .. import DirectConstraint
+from ..expressions.core import Expression, Comparison, Operator, BoolVal
+from ..expressions.globalconstraints import GlobalConstraint
+from ..expressions.globalfunctions import GlobalFunction
+from ..expressions.variables import _BoolVarImpl, NegBoolView, _IntVarImpl, _NumVarImpl
+from ..expressions.utils import is_num, is_any_list, eval_comparison, argval, argvals, get_bounds
+from ..transformations.get_variables import get_variables
+from ..transformations.normalize import toplevel_list
+from ..transformations.decompose_global import decompose_in_tree
+
+"""
+    Interface to CP Optimizers API
+
+    CP Optimizer, also a feature of IBM ILOG Optimization Studio, is a software library of constraint programming tools 
+    supporting constraint propagation, domain reduction, and highly optimized solution search.
+
+    Documentation of the solver's own Python API: (all modeling functions)
+    https://ibmdecisionoptimization.github.io/docplex-doc/cp/docplex.cp.modeler.py.html#module-docplex.cp.modeler
+
+    ===============
+    List of classes
+    ===============
+
+    .. autosummary::
+        :nosignatures:
+
+        CPM_cpo
+"""
+
+class CPM_cpo(SolverInterface):
+    """
+    Interface to CP Optimizers API
+
+    Requires that the 'docplex' python package is installed:
+    $ pip install docplex
+
+    docplex documentation:
+    https://ibmdecisionoptimization.github.io/docplex-doc/
+
+    You will also need to install CPLEX Optimization Studio from IBM's website.
+    There is a free community version available.
```

Comment:
maybe add 'and add the location of the cpoptimizer binary to your path'?

I did not think it would be possible that if you do a local pip install, you would still have to add something to the path... but hey, turns out we do ( so your original installer check was correct, I perhaps just tested the MP one instaed of the CP one...)

Reply:
should I revert to the other check? then we can also mention the path in the error message.
I think it just worked on my machine by installing the cplex studio